tools/SendBlobs/FundsDistributor.cs

```diff
@@ -0,0 +1,192 @@
+            throw new Exception($"Not enough balance on {distributeFrom.Address} to distribute to {keysToMake} addresses");
+
+        UInt256 balanceToDistribute = balanceMinusBuffer - totalFee;
+
+        UInt256 perKeyToSend = balanceToDistribute / keysToMake;
+
+        using PrivateKeyGenerator generator = new();
+        IEnumerable<PrivateKey> privateKeys = Enumerable.Range(1, (int)keysToMake).Select(i => generator.Generate()).ToArray();
+
+        List<string> txHash = new List<string>();
+
+        TxDecoder txDecoder = new();
+
+        foreach (PrivateKey key in privateKeys)
+        {
+            if (maxFee == 0)
+            {
+                gasPriceRes = await nodeManager.Post<string>("eth_gasPrice") ?? "1";
+                gasPrice = (UInt256)Convert.ToUInt64(gasPriceRes, gasPriceRes.StartsWith("0x") ? 16 : 10);
+
+                maxPriorityFeePerGasRes = await nodeManager.Post<string>("eth_maxPriorityFeePerGas") ?? "1";
+                maxPriorityFeePerGas = (UInt256)Convert.ToUInt64(maxPriorityFeePerGasRes, maxPriorityFeePerGasRes.StartsWith("0x") ? 16 : 10);
+            }
+
+            Transaction tx = CreateTx(chainId,
+                                      key.Address,
+                                      maxFee != 0 ? maxFee : gasPrice + maxPriorityFeePerGas,
+                                      nonce,
+                                      maxPriorityFeePerGas,
+                                      perKeyToSend);
+
+            await distributeFrom.Sign(tx);
+
+            string txRlp = Hex.ToHexString(txDecoder
+                .Encode(tx, RlpBehaviors.SkipTypedWrapping | RlpBehaviors.InMempoolForm).Bytes);
+
+            string? result = await nodeManager.Post<string>("eth_sendRawTransaction", "0x" + txRlp);
+            if (result != null)
+                txHash.Add(result);
+            nonce++;
+        }
+        if (!string.IsNullOrWhiteSpace(keyFilePath))
+            File.WriteAllLines(keyFilePath, privateKeys.Select(k => k.ToString()), Encoding.ASCII);
+
+        return txHash;
+    }
+    /// <summary>
+    /// Send all available funds from a list of private keys contained in <paramref name="keyFilePath"/> to <paramref name="beneficiary"/>.
+    /// </summary>
+    /// <param name="nodeManager"></param>
+    /// <param name="chainId"></param>
+    /// <param name="beneficiary"></param>
+    /// <param name="keyFilePath"></param>
+    /// <param name="logManager"></param>
+    /// <returns><see cref="IEnumerable{string}"/> containing all the executed tx hashes.</returns>
+    public async static Task<IEnumerable<string>> ReclaimFunds(INodeManager nodeManager, ulong chainId, Address beneficiary, string keyFilePath, ILogManager logManager, UInt256 maxFee, UInt256 maxPriorityFee)
+    {
+        IEnumerable<Signer> privateSigners =
+            File.ReadAllLines(keyFilePath, Encoding.ASCII)
+            .Select(k => new Signer(chainId, new PrivateKey(k), logManager));
```

Comment:
Generally line-by-line has lower memory footprint

Reply:
@LukaszRozmej, your comment is valid. IMO, it's just a tool for sending blobs, not a node code, so memory footprint is not that important here. Of course, if the purpose of the comment was to teach @ak88, it makes sense :)

Reply:
Since its only a test tool, i think it's okay to keep it as is.